terragrunt.hcl

```diff
@@ -0,0 +1,38 @@
+locals {
+  base = split("/", get_original_terragrunt_dir())
+#  common_vars
+  env_name = element(local.base, index(local.base, "environments") + 1)
+  env = yamldecode(file("${get_repo_root()}/environments/${local.env_name}/env.yaml"))
+}
+
+inputs = {
+  environment = local.env_name
+}
+
+generate "provider" {
+  path = "generated_provider.tf"
+  if_exists = "overwrite"
+  contents = <<EOF
+provider "aws" {
+  region = "${local.env.aws_region}"
+
+  default_tags {
+    tags = {
+      system = "kuberoke"
```

Comment:
all resources created by this module should use this tag (some exceptions may apply ™️ )